providers/core.tcpsocket.js

```diff
@@ -172,12 +172,12 @@ Socket_chrome.prototype.prepareSecure = function(cb) {
 };
 
 /**
- * Pauses the socket
- * @method pause
+ * Pauses the socket.  Internal-only
+ * @method pausePromise
  * @private
  * @returns {Promise} Promise to be fulfilled on pause.
  */
-Socket_chrome.prototype.pause = function() {
+Socket_chrome.prototype.pausePromise = function() {
```

Comment:
Can we have the one, non-promise-based implementation of pause and resume (since the underlying setPaused does this already), and change prepareSecure to use that one?